File: backend/app/models/schemas/document_category.py
"""Document-category related schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class DocumentCategoryCreate(BaseModel):
    """Create a category inside a knowledge base."""

    knowledge_base_id: int = Field(..., description="Owning knowledge base id")
    name: str = Field(..., min_length=1, max_length=100, description="Category name")
    parent_id: int | None = Field(None, description="Parent category id for subfolders")


class DocumentCategoryUpdate(BaseModel):
    """Rename a document category."""

    name: str = Field(..., min_length=1, max_length=100, description="Category name")


class DocumentCategoryResponse(BaseModel):
    """Document-category response."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    knowledge_base_id: int
    name: str
    parent_id: int | None = None
    document_count: int = 0
    created_at: datetime
    updated_at: datetime


class DocumentCategoryTreeNode(BaseModel):
    """A category node for tree display."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    knowledge_base_id: int
    name: str
    parent_id: int | None = None
    document_count: int = 0
    children: list["DocumentCategoryTreeNode"] = []
    created_at: datetime
    updated_at: datetime

File: frontend/src/shared/types/document-category.ts
export interface DocumentCategorySummary {
  id: number;
  knowledge_base_id: number;
  name: string;
  parent_id: number | null;
  document_count: number;
  created_at: string;
  updated_at: string;
}

export interface DocumentCategoryTreeNode {
  id: number;
  knowledge_base_id: number;
  name: string;
  parent_id: number | null;
  document_count: number;
  children: DocumentCategoryTreeNode[];
  created_at: string;
  updated_at: string;
}
